src/component/common/ConstraintAccordion/ConstraintAccordionEdit/StyledToggleButton/CaseSensitiveButton/CaseSensitiveButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { Tooltip } from '@mui/material';
+import { ReactComponent as CaseSensitive } from 'assets/icons/24_Text format.svg';
+import { ReactComponent as CaseSensitiveOff } from 'assets/icons/24_Text format off.svg';
+import React from 'react';
+import {
+    StyledToggleButtonOff,
+    StyledToggleButtonOn,
+} from '../StyledToggleButton';
+import { ConditionallyRender } from '../../../../ConditionallyRender/ConditionallyRender';
+import { IConstraint } from '../../../../../../interfaces/strategy';
+
+interface CaseSensitiveButtonProps {
+    localConstraint: IConstraint;
+    setCaseInsensitive: () => void;
+}
+
+export const CaseSensitiveButton = ({
+    localConstraint,
+    setCaseInsensitive,
+}: CaseSensitiveButtonProps) => {
+    return (
+        <ConditionallyRender
+            condition={Boolean(localConstraint.caseInsensitive)}
```

Comment:
Same with this condition

Reply:
typescript is complaining because caseInsensitive is boolean | undefined

Reply:
You're right. They're indeed marked as optional in the type. Disregard this comment.